### ISSUE TITLE
could_not_resolve_timezone when calling Timex.Timezone.get/1

### ISSUE THREAD

**Issue by jsmestad** [bitwalker/timex#653]
### Steps to reproduce

```elixir
    Tzdata.canonical_zone_list()
    |> Enum.map(&Timex.Timezone.get/1)
    |> Enum.map(fn %Timex.TimezoneInfo{full_name: full_name} = tz -> # fails here on match with {:error, ...}
      {"#{full_name} (UTC#{Timex.Timezone.total_offset(tz)})", full_name}
    end)
```

The error I am getting (running 3.6 and also tried `3.7.4` from git master)

```shell
     ** (ArgumentError) you attempted to apply :full_name on {:error, {:could_not_resolve_timezone, "Asia/Amman", 63783939421, :wall}}. If you are using apply/3, make sure the module is an atom. If you are using the dot syntax, such as map.field or module.function(), make sure the left side of the dot is an atom or a map
     stacktrace:
       :erlang.apply({:error, {:could_not_resolve_timezone, "Asia/Amman", 63783939421, :wall}}, :full_name, [])
       (heads_down 0.1.0) lib/heads_down_web/views/layout_view.ex:20: anonymous fn/1 in HeadsDownWeb.LayoutView.timezone_options_for_select/0
```

Also got this again in CI (after I thought it went away and closed the issue)

```shell
 The following arguments were given to anonymous fn/1 in HeadsDownWeb.LayoutView.timezone_options_for_select/0:
     
         # 1
         {:error, {:could_not_resolve_timezone, "Asia/Jerusalem", 63783943405, :wall}}
```

### Description of issue

- What are the expected results? 
 > This was not erroring until I tried it today so not sure if something with today is causing it or what. No code has changed from when this was working before. `tzdata` is `1.1.0`

I am not sure if this is somehow related to #555 but that is why I tried the latest release of Timex and head of master to see if that would fix things.

**Comment by jsmestad** [comment]
It looks like somehow this is no longer failing... 🤷

**Comment by bitwalker** [comment]
The issue is that what you are asking for with that API is a TimezoneInfo representing a current valid DateTime in that timezone, but the date/time given doesn't have a valid period based on the wall clock in that zone.

To make your use case easier, I've added an overload to that API, `Timex.Timezone.get/3` which allows you to specify `:utc` or `:wall` as the third argument to request a TimezoneInfo based on those respective clocks. In your case, what you want is `:utc`, since that will always have a valid corresponding period. As an example, you'd call it like so: `Timex.Timezone.get(DateTime.utc_now(), "Asia/Amman", :utc)`

**Comment by bitwalker** [comment]
This will be published in 3.7.5 btw

**Comment by jsmestad** [comment]
@bitwalker awesome! thank you 🙏
